proposals/20210817-firebase_ml_publisher_component.md

```diff
@@ -0,0 +1,45 @@
+#### SIG TFX-Addons
+# Project Proposal
+
+**Your name:** Chansung Park
+
+**Your email:** deep.diver.csp@gmail.com
+
+**Your company/organization:** Individual(ML GDE)
+
+**Project name:** [Firebase ML Publisher](https://github.com/tensorflow/tfx-addons/issues/59)
+
+## Project Description
+This project defines a custom TFX component to publish/update ML models from TFX Pusher to [Firebase ML](https://firebase.google.com/products/ml). The input model from TFX Pusher is assumed to be a TFLite format.
+
+## Project Category
+Component
+
+## Project Use-Case(s)
+This project helps users to publish trained models directly from TFX Pusher component to Firebase ML. 
+
+With Firebase ML, we can guarantee that mobile devices can be equipped with the latest ML model without explicitly embedding binary in the project compiling stage. We can even A/B test different versions of a model with Google Analytics when the model is published on Firebase ML.
+
+## Project Implementation
+Firebase ML Publisher component will be implemented as Python function-based component. You can find the [actual source code](https://github.com/sayakpaul/Dual-Deployments-on-Vertex-AI/blob/main/custom_components/firebase_publisher.py) in my personal project. 
+
+The implementation details
+- Define a custom Python function-based TFX component. It takes the following parameters from a previous component.
+  - The URI of the pushed model from TFX Pusher component.
```

Comment:
That is an option I guess? 

My understanding is that TFX Trainer just saved the trained model, and Pusher manages the different versions(correct me if I am wrong).

Since there are more after Trainer's job is done. For example, we don't know if the trained model is good enough, so we add Evaluator component with Resolver node. When the model passes thresholds defined in `eval_config`, the model is ready to be pushed.

Reply:
AFAIK, trainer could generate tflite models, and evaluator could handle tflite models.
https://www.tensorflow.org/tfx/tutorials/tfx/tfx_for_mobile

From pusher perspective, Pushed model type artifacts are for archiving already pushed models.
I think this new custom component should follow the standard pusher's interface instead of reading pusher's output, since this is another custom pusher. 

https://github.com/tensorflow/tfx/blob/921167ce1b53e7f0632bffacd20304cfed82d901/tfx/types/standard_component_specs.py#L284-L296

We don't have infra validator which supports tflite models for now, but we could have this in the future.

Reply:
Thank you for leaving the comment, and I think you are right. 

@nikelite 
```suggestion
  - It should follow the standard Pusher's interface.
```

@casassg please let me know your thought

Reply:
I agree w the conclusion reached. Let's make sure to modify the proposal w the new interface.